plugin/src/main/java/org/gradle/testretry/internal/executer/framework/TestFrameworkStrategy.java

```diff
@@ -25,25 +26,52 @@
 import org.gradle.util.GradleVersion;
 
 import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Iterator;
+import java.util.regex.Pattern;
 
 /**
  * Instances are scoped to a test task execution and are reused between rounds.
  */
 public interface TestFrameworkStrategy {
 
+    Pattern SPOCK2_CORE_JAR_NAME_PATTERN = Pattern.compile(".*/spock-core-2.*\\.jar");
```

Comment:
```suggestion
    Pattern SPOCK2_CORE_JAR_NAME_PATTERN = Pattern.compile(".*/spock-core-2[^/]*\\.jar");
```

Reply:
Done in https://github.com/gradle/test-retry-gradle-plugin/pull/202/commits/e03e80d0e4981334234d61187dc28f19da1ededb